packages/data/lib/src/request/api_request_representable.dart

```diff
@@ -0,0 +1,34 @@
+enum HTTPMethod { get, post, delete, put, patch }
+
+extension HTTPMethodString on HTTPMethod {
+  String get string {
+    switch (this) {
+      case HTTPMethod.get:
+        return "GET";
+      case HTTPMethod.post:
+        return "POST";
+      case HTTPMethod.delete:
+        return "DELETE";
+      case HTTPMethod.patch:
+        return "PATCH";
+      case HTTPMethod.put:
+        return "PUT";
+    }
+  }
+}
```

Comment:
had a fresh look, I don't think that this enum is a good solution. why? because it won't allow us to generalize http clients and their methods. what if another http client will use int consts for methods type, like

GET = 0;
POST = 1; and so on

or another client will have lower-case strings. I think that there's no need in this enum, better to use dio consts